eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -8,9 +8,60 @@ param (
 )
 Set-StrictMode -Version 3
 
+Install-Module -Name powershell-yaml -RequiredVersion 0.4.7 -Force -Scope CurrentUser
+
 . $PSScriptRoot/ChangedFiles-Functions.ps1
 . $PSScriptRoot/Logging-Functions.ps1
 
+function Find-Suppressions-Yaml {
```

Comment:
At a future point we might want to factor these out into common helpers but it is fine to be here for now.

Reply:
Given this logic will be duplicated in `openapi-alps` in TypeScript, I would appreciate if we extract all the logic related to handling `suppressions.yml` into its own file and slap `CODESYNC` comment on top of it, explaining the implementation in the file has to be kept in sync with other implementations. And also link to the related GitHub issue URL.

I assume here the CI pipeline will support it if we split the logic into its own file and dot-source / otherwise import it into `TypeSpec-Requirement.ps1`.

Reply:
My hope was that processing `suppressions.yml` would be so simple that duplicated code wouldn't be a problem, but I'm starting to second-guess this.  There are some complexities like how to process `path` that may be difficult to implement identically in multiple languages.

One option might be to implement this once as a PowerShell Function, and Node apps could call into PowerShell when needed?

Reply:
I'd suggest we at least start with a shared Powershell function and then we can decide how we want to handle this. Perhaps instead of having node call it we can process the suppressions and send them into the node tool from powershell.

Reply:
I will go ahead and extract these functions out into a separate powershell file in this PR.  I think it will help clarify our thinking about the design.

Reply:
Sorry to flip-flop, but Konrad and I discussed this at length.  Given the number of design decisions around reusing this code, I'm going to merge this PR with the internal implementation, to buy us more time for the bigger design.